plugins/user-settings/src/components/FeatureFlags/UserSettingsFeatureFlags.tsx

```diff
@@ -62,16 +63,25 @@ export const UserSettingsFeatureFlags = () => {
   return (
     <InfoCard title="Feature Flags">
       <List dense>
-        {featureFlags.map(featureFlag => {
+        <ListItem>
+          <TextField
+            label="Search"
+            onChange={e => setSearchInput(e.target.value)}
+            value={searchInput}
+          />
+        </ListItem>
+        {featureFlags.map((featureFlag, index) => {
           const enabled = Boolean(state[featureFlag.name]);
 
           return (
-            <FlagItem
-              key={featureFlag.name}
-              flag={featureFlag}
-              enabled={enabled}
-              toggleHandler={toggleFlag}
-            />
+            featureFlag.name.includes(searchInput) && (
```

Comment:
future improvement / over-engineering idea: `searchInput.split(/\s/).map(part => part.trim().toLowerCase()).filter(Boolean)` (outside of this `.map` loop) and then check that `featureFlag.name.toLowerCase()` contains ALL of those parts :)